addons/l10n_in_withholding/wizard/l10n_in_withhold_wizard.py

```diff
@@ -109,7 +109,8 @@ def _compute_company_id(self):
     def _compute_journal(self):
         for wizard in self:
             wizard.journal_id = wizard.company_id.l10n_in_withholding_journal_id or \
-                                wizard.env['account.journal'].search([('company_id', '=', wizard.company_id.id), ('type', '=', 'general')], limit=1)
+                                wizard.company_id.parent_id.l10n_in_withholding_journal_id or \
```

Comment:
I suppose you might have a chain of branches - in that case, I imagine you'd want to look from the most specific branch to the least specific parent?

This leverages the fact that 'parent_ids' are returned from root to child, and 'mapped' preserves the order of the original model's recordset when aggregating the comodel records.

```suggestion
            wizard.journal_id = wizard.company_id.parent_ids.l10n_in_withholding_journal_id[-1] or \

```

Reply:
IMO We shouldn't do this because `l10n_in_withholding_journal_id` is not a required field. If it is not set in parent and child, it will raise an `IndexError`.

Instead of this, we can use this
```python
wizard.company_id.parent_ids.filtered('l10n_in_withholding_journal_id')[:1].l10n_in_withholding_journal_id or \
```
Is that okay?

Reply:
Sorry - indeed, accessing `[-1]` will fail if the recordset is empty!

You could just do a `[:-1]` slice instead :smile: 

```suggestion
            wizard.journal_id = wizard.company_id.parent_ids.l10n_in_withholding_journal_id[:-1] or \
```

I think it's important to get the last element of the recordset: the `parent_ids` are returned in root-to-branch order, so the `parent_ids.l10n_in_withholding_journal_id` will be as well.